be/src/exprs/expr.cpp

```diff
@@ -351,12 +351,13 @@ Status Expr::create_vectorized_expr(starrocks::ObjectPool* pool, const starrocks
             *expr = pool->add(VectorizedCastExprFactory::from_thrift(
                     pool, texpr_node, (state == nullptr) ? false : state->query_options().allow_throw_exception));
             if (*expr == nullptr) {
-                LogicalType to_type = TypeDescriptor::from_thrift(texpr_node.type).type;
-                LogicalType from_type = thrift_to_type(texpr_node.child_type);
-                std::string err_msg = fmt::format(
-                        "Vectorized engine does not support the operator, cast from {} to {} failed, maybe use switch "
-                        "function",
-                        type_to_string_v2(from_type), type_to_string_v2(to_type));
+                TypeDescriptor to_type = TypeDescriptor::from_thrift(texpr_node.type);
+                TypeDescriptor from_type(thrift_to_type(texpr_node.child_type));
+                if (texpr_node.__isset.child_type_desc) {
+                    from_type = TypeDescriptor::from_thrift(texpr_node.child_type_desc);
+                }
```

Comment:
can you add a comment in the code, when this is used? is it legacy?

Reply:
ok